### PR TITLE
refactor: Updated 'Upgrade Requirements' workflow to use reusable workflow

### PR DESCRIPTION
**Pull request by Jawayria**
JIRA: https://openedx.atlassian.net/browse/BOM-2964

### CODE REVIEW

**Review by mraarif — APPROVED** [review]

File: .github/workflows/upgrade-python-requirements.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        uses: dawidd6/action-send-mail@v3
+        with:
+          server_address: email-smtp.us-east-1.amazonaws.com
+          server_port: 465
+          username: ${{secrets.edx_smtp_username}}
+          password: ${{secrets.edx_smtp_password}}
+          subject: Upgrade python requirements workflow executed successfully in ${{github.repository}}
+          to: ${{ inputs.email_address }}
+          from: github-actions <github-actions@edx.org>
+          body: Upgrade python requirements workflow executed in {{github.repository}}. For details see "github.com/${{ github.repository }}/pulls"
```
Comment (matthugs) [inline .github/workflows/upgrade-python-requirements.yml:101]:
```suggestion
          body: Upgrade python requirements workflow executed in ${{github.repository}}. For details see "github.com/${{ github.repository }}/pulls"
```

File: workflow-templates/upgrade-python-requirements.yml [hunk 2]
```diff
@@ -2,67 +2,25 @@ name: Upgrade Requirements
...
   schedule:
-    # will start the job at midnight every monday (UTC), should be updated to use appropriate value
     - cron: "0 0 * * 1"
   workflow_dispatch:
     inputs:
       branch:
         description: "Target branch to create requirements PR against"
         required: true
-        default: 'master'
+        default: "master"
```
Comment (matthugs) [inline workflow-templates/upgrade-python-requirements.yml:11]: you can use `$default_branch` here according to https://docs.github.com/en/actions/learn-github-actions/creating-workflow-templates

**Review by matthugs — APPROVED** [review]
I'd love to trade reviews if feasible for a feature I added to the underlying script: https://github.com/edx/testeng-ci/pull/274

**Comment by Jawayria** [comment]
> I'd love to trade reviews if feasible for a feature I added to the underlying script: [edx/testeng-ci#274](https://github.com/edx/testeng-ci/pull/274)

Yes Sure

**Pull request merged**